src/qt/bitcoin.cpp

```diff
@@ -459,8 +459,11 @@ int GuiMain(int argc, char* argv[])
     SetupUIArgs();
     std::string error;
     if (!node->parseParameters(argc, argv, error)) {
+        InitError(strprintf("Error parsing command line arguments: %s\n", error));
+        // Create a message box, because the gui has neither been created nor has subscribed to core signals
```

Comment:
It might be good to note in the comment on line 427 that the `handleMessageBox` call there is necessary for these `InitError` calls below to work:

https://github.com/bitcoin/bitcoin/blob/4fcccdac785e09ad5627b3bf4811dfba353693e8/src/qt/bitcoin.cpp#L427-L428

It'd also be possible to ensure handlers were properly registered inside `InitError` by adding an assert there:

```c++
assert(g_ui_signals.ThreadSafeMessageBox.num_slots() > 0);
```

https://github.com/bitcoin/bitcoin/blob/4fcccdac785e09ad5627b3bf4811dfba353693e8/src/ui_interface.cpp#L58-L62

Reply:
How would that add any value? Boost already has this assertion built in:

```
terminate called after throwing an instance of 'boost::wrapexcept<boost::signals2::no_slots_error>'
  what():  boost::signals2::no_slots_error

Reply:
Oh, I didn't realize ThreadSafeMessageBox uses `last_value<bool>` so an exception will already be thrown. In this case the assert would be redundant.

https://www.boost.org/doc/libs/1_70_0/doc/html/boost/signals2/no_slots_error.html
https://www.boost.org/doc/libs/1_70_0/doc/html/boost/signals2/last_value.html
https://www.boost.org/doc/libs/1_70_0/doc/html/boost/signals2/last_valu_1_3_37_6_5_1_1_2.html